memcache/options.go

```diff
@@ -0,0 +1,55 @@
+package memcache
+
+func newOptions(opts ...Option) *Options {
+	o := &Options{
+		Alloc: &defaultAllocator{},
```

Comment:
Rather than creating a new `defaultAllocator` instance for each `newOptions` call, could we share a single instance?

Reply:
Since it's a pointer, if we always share the same instance then the default one will be overridden each time we apply options, no?

Reply:
Sorry, I should've been clearer: I was suggesting sharing a single `defaultAllocator` instance, not sharing a single `Options` instance

Reply:
Right. Sure, that should be shared.

Reply:
I've made this change.